tests/python_tests/test_sfpu_reduce.py

```diff
@@ -62,48 +65,54 @@ def test_sfpu_reduce(
     reduce_pool,
     negative_number,
     dimension_combinations,
-    add_top_row,
 ):
-    if negative_number and formats.input_format == DataFormat.UInt32:
+    if negative_number and (
+        formats.input_format in [DataFormat.UInt32, DataFormat.UInt16]
+        or reduce_pool == ReducePool.Max
+    ):
         pytest.skip(
             f"Skipping negative_numbers=True for unsigned format {formats.input_format}"
         )
 
     input_dimensions = dimension_combinations
     torch_format = format_dict[formats.input_format]
 
-    src_A, src_B, tile_cnt = generate_stimuli(
-        formats.input_format, formats.input_format, input_dimensions=input_dimensions
+    # STIMULI GENERATION
+    tile_cnt = input_dimensions[0] * input_dimensions[1] // 1024
+    max, min = 127, (
+        0 if formats.input_format in [DataFormat.UInt32, DataFormat.UInt16] else -127
     )
-    src_A = torch.ones(tile_cnt * 1024, dtype=torch_format)
+    src_A = torch.randint(
+        low=min, high=max, size=(tile_cnt * 1024,), dtype=torch_format
+    )
+    # src_A = torch.cat([torch.full((1024,), i, dtype=torch_format) for i in range(1, 5)])
+    src_B = torch.zeros_like(src_A)
 
-    # Generate 4 faces with all 2s for easy verification (column sums = 32*2 = 64, which is a multiple of 32)
     sign = -1 if negative_number else 1
     src_A *= sign
 
-    generate_golden = get_golden_generator(UnarySFPUGolden)
-    golden_tensor = generate_golden(
-        mathop,
-        src_A,
+    # Max Reduction can do block and single tile reduction whereas Sum/Avg only do single tile reduction, convert Sum/Avg golden to do block reduction by retilizing input to src_A
+    # Dimensions for Max reduction work column wise, for Sum/Avg processing tiles independently is same as column reduction on dst block dimension [32, num_tiles * 32] where num rows is 32 i.e RT_DIM=1 (same as a single tile)
+    dst_dim = input_dimensions
+    if reduce_pool != ReducePool.Max:
+        dst_dim = [32, tile_cnt * 32]
+    src_A = untilize_block(src_A, formats.input_format, dst_dim)
+    src_A = tilize_block(
+        src_A, dst_dim, stimuli_format=formats.input_format
+    ).flatten()  # Input tensor is tilized in dst register
+    src_A_untilized = untilize_block(
+        src_A, formats.input_format, dst_dim
+    )  # Passed into golden since PyTorch library has no concept of tilization
```

Comment:
```suggestion
    src_A_untilized = untilize_block(src_A, formats.input_format, input_dimensions)
```
Why can't we just untilize once and skip the tilize/flatten/untilize cycle?

Reply:
Yup, I had this here when I was debugging so I could visualize tilized inputs better. Will remove it 👍